src/shared/remoteSourceTrackingService.ts

```diff
@@ -458,7 +466,7 @@ ${formatSourceMemberWarnings(outstandingSourceMembers)}`
     const matchingKey = sourceMembers.get(key)
       ? key
       : getDecodedKeyIfSourceMembersHas({ sourceMembers, key, logger: this.logger });
-    this.sourceMembers.set(matchingKey, sourceMember);
+    this.sourceMembers.set(matchingKey, { ...sourceMember, MemberName: decodeURIComponent(sourceMember.MemberName) });
```

Comment:
check the logic here: let's always use a decoded key; the `get` already checks for the encoded version if it doesn't find a match

Reply:
I think this should be safe, both functionally and performance.